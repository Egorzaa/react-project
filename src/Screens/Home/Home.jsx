import { CounterWithLogger } from "../../Components/Counter";
import { Child } from "../../Components/Counter/Child";

import "./Home.css";

export const Home = () => {
  return (
    <div className="App">
      <header className="App-header">
        <CounterWithLogger render={() => <h1>Test render props pattern</h1>}>
          {({ color }, props) => {
            console.log("rerender");
            return (
              <div style={{ color }} {...props}>
                counter text <input />
              </div>
            );
          }}
        </CounterWithLogger>
        <Child />
      </header>
    </div>
  );
};
