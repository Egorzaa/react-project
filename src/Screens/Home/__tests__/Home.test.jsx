import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import { Home } from "../Home";
import { store } from "../../../Store";

describe("test Home component", () => {
  it("find 'learn react' text", () => {
    render(
      <Provider store={store}>
        <Home />
      </Provider>
    );
    const linkElement = screen.getByText(/learn react/i);
    expect(linkElement).toBeInTheDocument();
  });
  it("find 'Test render props pattern' text", () => {
    render(
      <Provider store={store}>
        <Home />
      </Provider>
    );
    const linkElement = screen.getByText(/Test render props pattern/i);
    expect(linkElement).toBeInTheDocument();
  });
  it("find 'Test render props pattern' text", () => {
    const component = render(
      <Provider store={store}>
        <Home />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
