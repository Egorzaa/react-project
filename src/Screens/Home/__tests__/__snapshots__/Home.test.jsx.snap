// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Home component find 'Test render props pattern' text 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header
          class="App-header"
        >
          <img
            alt="logo"
            class="App-logo"
            src="logo.svg"
          />
          <h1>
            Vasja
          </h1>
          <p>
            Edit 
            <code>
              src/App.js
            </code>
             and save to reload.
          </p>
          <a
            class="App-link"
            href="https://reactjs.org"
            rel="noopener noreferrer"
            target="_blank"
          >
            Learn React!
          </a>
          <div>
            <div>
              0
            </div>
            <div>
              previous 
            </div>
            <div>
              <input
                value=""
              />
            </div>
            <div>
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-error Mui-focused css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="standard-basic"
                  id="standard-basic-label"
                >
                  Standard
                </label>
                <div
                  class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-error Mui-focused MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
                >
                  <input
                    aria-invalid="true"
                    class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                    id="standard-basic"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <button>
              Click me!
            </button>
            <div>
              <div
                style="color: red;"
              >
                counter text 
                <input />
              </div>
            </div>
            <div>
              <h1>
                Test render props pattern
              </h1>
            </div>
          </div>
          <div>
            1
          </div>
          <div>
            2
          </div>
          <div>
            3
          </div>
          <div>
            4
          </div>
          <div>
            Vasja
          </div>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header
        class="App-header"
      >
        <img
          alt="logo"
          class="App-logo"
          src="logo.svg"
        />
        <h1>
          Vasja
        </h1>
        <p>
          Edit 
          <code>
            src/App.js
          </code>
           and save to reload.
        </p>
        <a
          class="App-link"
          href="https://reactjs.org"
          rel="noopener noreferrer"
          target="_blank"
        >
          Learn React!
        </a>
        <div>
          <div>
            0
          </div>
          <div>
            previous 
          </div>
          <div>
            <input
              value=""
            />
          </div>
          <div>
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-error Mui-focused css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for="standard-basic"
                id="standard-basic-label"
              >
                Standard
              </label>
              <div
                class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-error Mui-focused MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
              >
                <input
                  aria-invalid="true"
                  class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                  id="standard-basic"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <button>
            Click me!
          </button>
          <div>
            <div
              style="color: red;"
            >
              counter text 
              <input />
            </div>
          </div>
          <div>
            <h1>
              Test render props pattern
            </h1>
          </div>
        </div>
        <div>
          1
        </div>
        <div>
          2
        </div>
        <div>
          3
        </div>
        <div>
          4
        </div>
        <div>
          Vasja
        </div>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
